### PR TITLE
Improved the way how the article slug is generated

### PR DESCRIPTION
**Pull request by takeit**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-997
| License                   | AGPLv3

### CODE REVIEW

File: features/02_articles/validating_article_slugline.feature [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  As a HTTP Client
+  I want to be able to check if the article slug was generated properly
+
+  Scenario: Submitting request payload in ninjs format and publishing a package
+    Given I am authenticated as "test.user"
+    When I add "Content-Type" header equal to "application/json"
+    And I send a "POST" request to "/api/{version}/content/push" with body:
+    """
+      {"language": "en", "body_html": "<p>some html body</p>", "versioncreated": "2016-09-23T13:57:28+0000", "firstcreated": "2016-09-23T09:11:28+0000", "description_text": "some abstract text", "place": [{"country": "Australia", "world_region": "Oceania", "state": "Australian Capital Territory", "qcode": "ACT", "name": "ACT", "group": "Australia"}], "version": "2", "byline": "ADmin", "keywords": [], "guid": "urn:newsml:localhost:2016-09-23T13:56:39.404843:56465de4-0d5c-495a-8e36-3b396def3cf0", "priority": 6, "subject": [{"name": "lawyer", "code": "02002001"}], "urgency": 3, "type": "text", "headline": "Abstract html test", "service": [{"name": "Australian General News", "code": "a"}], "description_html": "<p><b><u>some abstract text</u></b></p>", "located": "Sydney", "pubstatus": "usable"}
```
Comment (ahilles107) [inline features/02_articles/validating_article_slugline.feature]: can't we pretty format it? It's easier to see what's pushed in test.

File: features/02_articles/validating_article_slugline.feature [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        "route":{
+          "name":"article",
+          "type":"content"
+        }
+      }
+     """
+    Then the response status code should be 201
+    And I am authenticated as "test.user"
+    And I add "Content-Type" header equal to "application/json"
+    Then I send a "POST" request to "/api/{version}/packages/5/publish/" with body:
```
Comment (ahilles107) [inline features/02_articles/validating_article_slugline.feature]: how do You know that it's fifth? First 4 are loaded in fixtures?

File: features/content_push/validating_payload_without_slugline_metadata.feature [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  In order to publish an article
+  As a HTTP Client
+  I want to be able to check if the request payload is valid according to ninjs schema
+
+  Scenario: Submitting request payload in ninjs format
+    Given I am authenticated as "test.user"
+    When I add "Content-Type" header equal to "application/json"
+    And I send a "POST" request to "/api/{version}/content/push" with body:
+    """
+      {"language": "en", "body_html": "<p>some html body</p>", "versioncreated": "2016-09-23T13:57:28+0000", "firstcreated": "2016-09-23T09:11:28+0000", "description_text": "some abstract text", "place": [{"country": "Australia", "world_region": "Oceania", "state": "Australian Capital Territory", "qcode": "ACT", "name": "ACT", "group": "Australia"}], "version": "2", "byline": "ADmin", "keywords": [], "guid": "urn:newsml:localhost:2016-09-23T13:56:39.404843:56465de4-0d5c-495a-8e36-3b396def3cf0", "priority": 6, "subject": [{"name": "lawyer", "code": "02002001"}], "urgency": 3, "type": "text", "headline": "Abstract html test", "service": [{"name": "Australian General News", "code": "a"}], "description_html": "<p><b><u>some abstract text</u></b></p>", "located": "Sydney", "pubstatus": "usable"}
```
Comment (ahilles107) [inline features/content_push/validating_payload_without_slugline_metadata.feature]: again - let's make it multiline.

File: features/02_articles/validating_article_slugline.feature (hunk 2 above)
Comment (takeit) [reply features/02_articles/validating_article_slugline.feature]: yes

Comment (takeit) [reply features/02_articles/validating_article_slugline.feature]: preferably, the database should be empty by default, and we should create all needed resources using contexts, like creating a new tenant (`Given there is a new tenant ...`) etc. This way we would have more control over the data that is inserted into the database in the given scenario. But if the current implementation is ok we can stick to it for now.

**Comment by takeit** [comment]
Travis has some issues, but in general, it passes the tests there so it can be merged IMO.

**Pull request merged**